### PR TITLE
New canary

### PR DESCRIPTION
**Pull request by diocas**

### CODE REVIEW

**Comment by diocas** [comment]
Piggy backed other small fixes together with the canary changes.

**Comment by diocas** [comment]
Separated the public link fixes from the canary stuff to allow me to deploy it immediately.

File: api/canary/canary.go [hunk 1]
```diff
@@ -25,51 +26,50 @@ func New(opts *Options) *Manager {
...
 
-func (cm *Manager) SetStatus(username string, adopter bool) error {
-	isAdopter := "no"
-	if adopter {
-		isAdopter = "yes"
+func (cm *Manager) SetVersion(username, version string) error {
+
+	valid := Find([]string{"production", "canary", "ocis"}, version)
+	if !valid {
+		errors.New("Invalid option")
```
Comment (labkode) [inline api/canary/canary.go:51]: return here and log the invalid option given

Comment (diocas) [reply api/canary/canary.go:51]: ups..

**Review by labkode — CHANGES_REQUESTED** [review]
Just that error handling, the rest looks good.

File: api/canary/canary.go (hunk 1 above)
Comment (diocas) [reply api/canary/canary.go]: Fixed! the log happens in ocproxy.

**Comment by diocas** [comment]
Can merge

**Review by labkode — APPROVED** [review]

**Pull request merged**